.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,132 @@
+    steps:
+      - name: Checkout PR commit
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - id: set-matrix
+        name: Calculate Matrix
+        run: |
+          echo "php_versions=$(jq -c .services.php supported-services.json)" >> "$GITHUB_OUTPUT"
+
+      - name: Debug output
+        run: |
+          echo "PHP Versions: ${{ steps.set-matrix.outputs.php_versions }}"
+    
+  unit-tests:
+    needs: matrix-calculator
+    strategy:
+      fail-fast: false
+      matrix:
+        php_version: ${{ fromJSON(needs.matrix-calculator.outputs.php_versions) }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout PR commit
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          path: main
+          # Need to do this otherwise Nx cant determine diff
+          fetch-depth: 0
+
+      - name: Fetch base
+        working-directory: ./main
+        run: git fetch origin ${{ github.event.pull_request.base.ref }}
+
+      - name: Install Composer
+        uses: php-actions/composer@v6
+        with:
+          # Run the dummy help command after installing Composer (otherwise defaults to install)
+          command: help
+
+      - name: Get Composer Cache Directory
+        id: composer-cache
+        working-directory: ./main
+        run: echo "composer_cache_dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
+
+      - name: Install Composer Cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.composer-cache.outputs.composer_cache_dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/main/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-composer-
+
+      - name: Install Composer Dependencies
+        working-directory: ./main
+        run: composer install
+
+      - name: Checkout Nx Repo
+        uses: actions/checkout@v3
+        with:
+          repository: adamzero1/nx-for-php
+          ref: docker-wrapper
+          path: nx
+
+      - name: Install / setup Nx
+        working-directory: ./main
+        run: |
+          npm install -g nx@15.4.1
+          NXDIR="../nx"
+          cp -r ${NXDIR}/nx ./ 
+          cp ${NXDIR}/nx.json ./ 
+          cp ${NXDIR}/package.json ./ 
+          cp ${NXDIR}/package-lock.json ./ 
+          npm ci
+          npm run generate-workspace
+
+      - name: Print Affected
+        working-directory: ./main
+        run: |
+          AFFECTED_OUTPUT=/tmp/affect.json
+          nx print-affected --head=HEAD --base=remotes/origin/${{ github.event.pull_request.base.ref }} > ${AFFECTED_OUTPUT}
+          echo "Affected Projects: $(jq .projects ${AFFECTED_OUTPUT})"
+      
+      # Warden / Den stuff
+      - name: Checkout Den Repo
+        uses: actions/checkout@v3
+        with:
+          repository: swiftotter/den
+          path: den
+
+      - name: Init / Configure Den

```

Comment:
Would it make sense to switch to warden already, as den is kind of deprecated?

Reply:
@Vinai do you mean the reverse? Den is the new, warden is the old.

Reply:
No, @Vinai is correct: https://github.com/orgs/wardenenv/discussions/556

Reply:
What a confusing situation.

Reply:
I know, haha never mind.
This should be good now. 
I've double checked and it appears to be working.